cli/dcoscli/node/main.py

```diff
@@ -65,12 +67,324 @@ def _cmds():
+    """
+    def _is_success(status_code):
+        # consider 400 and 503 to be successful status codes.
+        # API will return the error message.
+        if status_code in [200, 400, 503]:
+            return True
+        return False
+
+    # POST to snapshot api
+    client = mesos.DCOSClient()
+    url = client.get_dcos_url(url)
+    if method.lower() == 'get':
+        http_response = http.get(url, is_success=_is_success, timeout=timeout,
+                                 **kwargs)
+    elif method.lower() == 'post':
+        http_response = http.post(url, is_success=_is_success, timeout=timeout,
+                                  stream=stream, **kwargs)
+    else:
+        raise DCOSException('Unsupported HTTP method: ' + method)
+    return http_response
+
+
+def _do_snapshot_request(url, method, **kwargs):
+    """
+    Make HTTP request and expect a JSON response.
+
+    :param url: url
+    :type url: string
+    :param method: HTTP method, GET or POST
+    :type method: string
+    :return: snapshot JSON repsponse
+    :rtype: dict
+    """
+    http_response = _do_request(url, method, **kwargs)
+    return _read_http_response_body(http_response)
+
+
+def _read_http_response_body(http_response):
+    """
+    Get an requests HTTP response, read it and deserialize to json.
+
+    :param http_response: http response
+    :type http_response: requests.Response onject
+    :return: deserialized json
+    :rtype: dict
+    """
+    data = b''
+    try:
+        for chunk in http_response.iter_content(1024):
+            data += chunk
+        snapshot_response = util.load_jsons(data.decode('utf-8'))
+        return snapshot_response
+    except DCOSException:
+        raise
+
+
+def _confirm(prompt):
+    """
+    :param prompt: message to display to the terminal
+    :type prompt: str
+    :returns: True if the user responded with yes; False otherwise
+    :rtype: bool
+    """
+    while True:
+        sys.stdout.write('{} [yes/no] '.format(prompt))
+        sys.stdout.flush()
+        response = sys.stdin.readline().strip().lower()
+        if response == 'yes' or response == 'y':
+            return True
+        elif response == 'no' or response == 'n':
+            return False
+        else:
+            emitter.publish(
+                "'{}' is not a valid response.".format(response))
+
+
+def _snapshot_download(snapshot, to):
+    """
+    Download snapshot and put in the the current directory.
+
+    :param snapshot: snapshot file name.
+    :type snapshot: string
+    :return: exit code
+    :rtype: int
+    """
+
+    # make sure the requested snapshot exists
+    snapshot_size = 0
+    for available_snapshot in _get_snapshots_list():
+        assert len(available_snapshot) == 2, (
+            'Incorrect snapshot response'
+        )
+        if available_snapshot[0] == snapshot:
+            snapshot_size = available_snapshot[1]
+
+    url = '/system/health/v1/report/snapshot/serve/' + snapshot
```

Comment:
can we store the base and just append the specific request? Maybe we can have a snapshot class with all the shared info?